crates/terminal/src/terminal.rs

```diff
@@ -320,10 +320,6 @@ impl TerminalBuilder {
         window: AnyWindowHandle,
         completion_tx: Sender<()>,
     ) -> Result<TerminalBuilder> {
-        // TODO: Properly set the current locale,
-        env.entry("LC_ALL".to_string())
-            .or_insert_with(|| "en_US.UTF-8".to_string());
-
```

Comment:
I think we might need this on macOS. On macOS, if Zed is launched from `.app`, then `LC_ALL` might not be set and we need to call `setlocale` after reading `NSLocale` and setting `LANG`.

This here is essentially a workaround to `LC_*` not being set on macOS when launched as an `.app`, if I understand the code correctly.  See here: https://github.com/rogual/neovim-dot-app/issues/54